### ISSUE TITLE
Get docs working on cljdoc

### ISSUE THREAD

**Issue by lread** [clojure-emacs/refactor-nrepl#380]
## Problem
Docs for refactor-nrepl are failing to build on cljdoc, example: 
https://cljdoc.org/d/refactor-nrepl/refactor-nrepl/3.5.2

## Steps already taken
@vemv has already asked cljdoc to only analyze for Clojure, this will help.

### Lein plugin
Cljdoc loads namespaces to discover the API. 
Leiningen is not present and therefore analysis of the lein plugin will fail.
If I include `leiningen-core` as a `provided` dependency in refactor-nrepl's `project.clj`, cljdoc will include it and plugin analysis will succeed.
So that's good.

### What public API should refactor-nrepl expose on cljdoc?
After adjusting for leningen, a local test exposes the following namespaces:

![image](https://user-images.githubusercontent.com/967328/160025487-2ac17259-e4e9-46d7-9038-10b211a25a12.png)

Which seems like a bit much, right? 
I assume `inlined-deps` should not be documented.
What would be the API you would expect to be exposed on cljdoc?
Is the rest part of your supported public API?

**Comment by bbatsov** [comment]
> (Since this repo links to cljdoc via a cljdoc badge, I assume that you'd prefer they work)

Yeah, for sure. For various reasons all of our projects have been kind of cursed and don't work with cljdoc (e.g. cider-nrepl and orchard don't work as well), which is a pity.
